ygen/gogen.go

```diff
@@ -2146,6 +2149,15 @@ func findMapPaths(parent *Directory, fieldName string, compressPaths, shadowSche
 		// resolved it the field that the leafref points to. So, we
 		// compare their absolute paths for equality.
 		if k.Parent.Parent.Dir[k.Name].Type.Kind == yang.Yleafref && cmp.Equal(util.SchemaPathNoChoiceCase(k), fieldSlicePath) {
+			if util.IsConfig(k.Parent.Parent.Dir[k.Name]) != (util.StripModulePrefix(childPath[len(childPath)-2]) == "config") {
+				// The key node is a second path to this leaf if both are either config or state.
+				// For example, if the leafref is a config value, then it must refer to the config
+				// version of the child leaf, and vice versa for a state value, in order to pair the key
+				// with the leaf. The reason why leafref's "Path" field cannot be used
```

Comment:
What is the transformation that `TransformEntry` does to the `Path` field here that makes it unreliable?

Reply:
`TransformEntry` mutates the `Path` field by changing from `config` to `state` in its path string. This happens if `PreferOperationalState=true` (see https://github.com/openconfig/ygot/pull/424). It seemed that in one or more places, the code depended on `Path` actually referring to the "prioritized" sibling, so we can't use this field to determine to which one the leafref points.